### DISCUSSION TITLE
Information of the network (peers, channel and etc...)

### DISCUSSION THREAD

**Discussion by douglasjava** [hyperledger/fabric-gateway#597] [Q&A]
how is it possible to get a list of the peers or informations of the network, which applications have these options

is it possible?

**Comment by bestbeforetoday** [comment]
To perform business operations (evaluate and submit transactions, receive events), identifying all the network nodes is not necessary as the peer Gateway service deals with all this on the client's behalf. Therefore the Fabric Gateway client API does not provide this (or any other administrative) capability.

Information on the (active) network peers can be obtained using service discovery. This can be accessed using a [gRPC](https://grpc.io/) service exposed by peers. The [Discovery](https://hyperledger.github.io/fabric-protos/protos.html#discovery-Discovery) service can be invoked using [published language bindings](https://hyperledger.github.io/fabric-protos/) (for Go, Node and Java) generated from the Fabric protocol buffer messages. You would need to send a [PeerMembershipQuery](https://hyperledger.github.io/fabric-protos/protos.html#discovery-Query) to the discovery service.

The legacy Fabric SDKs typically do use service discovery internally to build a client-side model of all the network nodes. You could use those legacy SDKs to obtain the network peers but it is a very heavyweight hammer to use to access the discovery service information that you could get with a single gRPC service invocation using the fabric-protos packages mentioned above.
